Baikal/RenderFactory/render_factory.h

```diff
@@ -26,12 +26,25 @@
 #include "CLW.h"
 #include "Controllers/scene_controller.h"
 
+
 namespace Baikal
 {
     class Renderer;
     class Output;
     class PostEffect;
-    
+
+#ifdef ENABLE_MLDENOISER
```

Comment:
There are too many `#ifdef`s. I propose to keep a single `#ifdef` inside the factory method (throw if the macro is undefined) and make the rest of definitions always visible.